packages/core/src/browser/shell/side-panel-handler.ts

```diff
@@ -41,6 +42,12 @@ export const SidePanelHandlerFactory = Symbol('SidePanelHandlerFactory');
 
 export const SIDE_PANEL_TOOLBAR_CONTEXT_MENU: MenuPath = ['SIDE_PANEL_TOOLBAR_CONTEXT_MENU'];
 
+export namespace SettingsMenu {
```

Comment:
It does not look good that the side panel handler now knows about what is added to it as well as about preference menus. We should refactor it, the handler should add tabs at the bottom. Such constants should go to common contribution as well as adding an action tab.

Reply:
Make it a `SidebarTabMenuContribution` and extensions can contribute tabs which can open menu? 
Just like VSCode's `Accounts` and `Manage`
![image](https://user-images.githubusercontent.com/5847142/90208894-8892e400-de1c-11ea-91cc-74373b986eaf.png)
![image](https://user-images.githubusercontent.com/5847142/90209122-15d63880-de1d-11ea-9f97-51c33c95bee2.png)

Reply:
I think we should improve `ApplicationShell.addWidget` method which allows to add widgets to the bottom of side bar. Then there should be a preference frontend contribution which inject preferences menu widget using this API. If someone wants to add another widget he should do the same.

Maybe we can do it by extending `ApplicationShell.Area` to contain `left-bottom` and `right-bottom`.

Reply:
The `SettingsMenuWidget` is an icon and opens a menu on clicks, it dose not open a panel. 
![](https://user-images.githubusercontent.com/5847142/90209122-15d63880-de1d-11ea-9f97-51c33c95bee2.png)
Panels have enough spaces for `Widget` to render.
But `left-bottom` do not enough spaces
![image](https://user-images.githubusercontent.com/5847142/90379463-5cda5d00-e0ad-11ea-83d1-429a85a4a5be.png)

So is `ApplicationShell.addWidget` appropriate?

Reply:
Should we restricts the behavior of sidebars? `left` or `right` open panels, `left-bottom` or `right-bottom` open menus
![image](https://user-images.githubusercontent.com/5847142/90380210-58fb0a80-e0ae-11ea-9329-16b15a2c80fb.png)

Reply:
ok, I see you are right. Let's do then following there should be 2 top-level menu paths `LEFT_BAR_MENU`  and `RIGHT_BAR_MENU`. Second-level menu paths should be rendered as bottom tab widgets in corresponding side bars. 3rd-level menu paths should be rendered as context menu actions.